### ISSUE TITLE
Is it necessary to add endUpdateCallback and endDrawingCallback feature?

### ISSUE THREAD

**Issue by xingbofeng** [cornerstonejs/cornerstoneTools#1007]
### Prerequisites

* [x] Are you running the latest version?
* [x] Are you reporting to the correct repository?
* [x] Did you perform a cursory search?

For more information, see the `CONTRIBUTING` guide.

### Description

I saw the [vNext](https://github.com/cornerstonejs/cornerstoneTools/projects/5), and the plan to #915 , and I have the problem that I want to know the new measurement added or the old measurement changed, so I add the `endUpdateCallback` and `endDrawingCallback` in `moveNewHandle.js`, `moveHandle.js` and `moveAllHandle.js`, I Have add my custom feature in my version(forked from the official), @dannyrb If you think it's necessary to add it , I will create PR.

for example I changed the moveNewHandle.js

https://github.com/cornerstonejs/cornerstoneTools/blob/710ae6d8310f650cb15a51d08e5ef74031c41c7b/src/manipulators/moveNewHandle.js#L176-L184

```javascript
  // Why would a Touch_Pinch or Touch_Press be associated with a new handle?
  if (evt.type === EVENTS.TOUCH_PINCH || evt.type === EVENTS.TOUCH_PRESS) {
    handle.active = false;
    external.cornerstone.updateImage(element);
    if (typeof options.doneMovingCallback === 'function') {
      options.doneMovingCallback();
    }

    // add endDrawingCallback
    const tool = getToolForElement(element, toolName);
    if (tool && tool.endDrawingCallback) {
      tool.endDrawingCallback(evt, annotation, 'mouse');
    }

    return;
  }
```

**Comment by dannyrb** [comment]
I'll need to think on this, but I'll try to get back to you early next we. Don't hesitate to `@mention` me if I'm lagging in a response.

**Comment by xingbofeng** [comment]
hhh, thank your reply, i will mention you!

**Comment by JamesAPetts** [comment]
Hey @ xingbofeng,

Thanks for the quesiton, just so we are on the same page:

I understand the intent of your `endDrawingCallback`, in cornerstone vocabulary I guess that would be a "`measurementCompletedCallback`. Maybe an example of your use-case would help clarify, but any tool complicated enough to need some clean up at the end of drawing (e.g. freehand annoations), likely wouldn't be using these generic helpers?

Also, what is the logic behind `endUpdateCallback`, how would its triggering conditions be different from `doneMovingCallback`? Could you give an example?

Thanks

**Comment by xingbofeng** [comment]
@JamesAPetts Yes, the `endDrawingCallback` is the `measurementCompletedCallback`, now i can see the event of `MEASUREMENT_COMPLETED` is added to the lastest version.

So, can you add `MEASUREMENT_END_MODIFIED` event to the lastest version.The example like this:

![](http://ww1.sinaimg.cn/large/005SpQEcgy1g58p2ivopkg31240o2gtc.gif)
